clusterman/migration/worker.py

```diff
@@ -131,4 +157,44 @@ def event_migration_worker(migration_event: MigrationEvent, worker_setup: Worker
     :param MigrationEvent migration_event: event instance
     :param WorkerSetup worker_setup: migration setup
     """
-    pass  # TODO
+    prescaling_applied = False
+    manager = PoolManager(migration_event.cluster, migration_event.pool, SUPPORTED_POOL_SCHEDULER, fetch_state=False)
+    connector = cast(KubernetesClusterConnector, manager.cluster_connector)
+    connector.set_label_selectors(migration_event.label_selectors, add_to_existing=True)
+    manager.reload_state()
+    try:
+        if worker_setup.disable_autoscaling:
+            logger.info(f"Disabling autoscaling for {migration_event.cluster}:{migration_event.pool}")
+            disable_autoscaling(
+                migration_event.cluster,
+                migration_event.pool,
+                SUPPORTED_POOL_SCHEDULER,
+                time.time() + worker_setup.expected_duration,
+            )
+        if worker_setup.prescaling:
+            nodes = manager.get_node_metadatas(AWS_RUNNING_STATES)
+            offset = worker_setup.prescaling.of(len(nodes))
+            avg_weight = mean(node.instance.weight for node in nodes)
+            original_capacity = manager.target_capacity
+            prescaled_capacity = round(manager.target_capacity + offset * avg_weight)
+            manager.modify_target_capacity(prescaled_capacity)
+            prescaling_applied = True
+        if not _monitor_pool_health(
+            manager, time.time() + INITIAL_POOL_HEALTH_TIMEOUT_SECONDS, drained=[], check_pods=False
+        ):
+            raise NodeMigrationError(f"Pool {migration_event.cluster}:{migration_event.pool} is not healthy")
+        node_selector = lambda node: node.agent.agent_id and not migration_event.condition.matches(node)  # noqa
+        migration_routine = partial(_drain_node_selection, manager, node_selector, worker_setup)
+        if not limit_function_runtime(migration_routine, worker_setup.expected_duration):
+            raise NodeMigrationError(f"Failed migrating nodes for event {migration_event}")
+    except Exception as e:
+        logger.error(f"Issue while processing migration event {migration_event}: {e}")
+        raise
+    finally:
+        if worker_setup.disable_autoscaling:
+            logger.info(f"Re-enabling autoscaling for {migration_event.cluster}:{migration_event.pool}")
+            enable_autoscaling(migration_event.cluster, migration_event.pool, SUPPORTED_POOL_SCHEDULER)
+        if worker_setup.prescaling and prescaling_applied and prescaled_capacity >= manager.target_capacity:
+            # Only restoring original capacity of the target remained equal or lower than
+            # our set target (i.e. the autoscaler didn't have to scale up in the meantime)
+            manager.modify_target_capacity(original_capacity)
```

Comment:
hmm, i wonder if it's worth leaving this out: presumably clusterman will run relatively soon after autoscaling is re-enabled so at most we'll spend an extra 5 minutes (i *think* that's the longest period we run?) scaled up?

Reply:
I was thinking this as well. Can't tell how sensible the autoscaler will be in certain cases, but I guess won't leave it running overscaled for days 🤞